pkg/cover/report.go

```diff
@@ -61,28 +63,28 @@ func (rg *ReportGenerator) Do(w io.Writer, pcs []uint64) error {
 	for i, pc := range pcs {
 		pcs[i] = PreviousInstructionPC(rg.arch, pc)
 	}
-	covered, prefix, err := rg.symbolize(pcs)
+	covered, err := rg.symbolize(pcs)
 	if err != nil {
 		return err
 	}
 	if len(covered) == 0 {
 		return fmt.Errorf("'%s' does not have debug info (set CONFIG_DEBUG_INFO=y)", rg.vmlinux)
 	}
 	uncoveredPCs := rg.uncoveredPcsInFuncs(pcs)
-	uncovered, prefix2, err := rg.symbolize(uncoveredPCs)
+	uncovered, err := rg.symbolize(uncoveredPCs)
 	if err != nil {
 		return err
 	}
-	if len(uncoveredPCs) != 0 {
-		prefix = combinePrefix(prefix, prefix2)
-	}
-	return rg.generate(w, prefix, covered, uncovered)
+	return rg.generate(w, covered, uncovered)
 }
 
-func (rg *ReportGenerator) generate(w io.Writer, prefix string, covered, uncovered []symbolizer.Frame) error {
+func (rg *ReportGenerator) generate(w io.Writer, covered, uncovered []symbolizer.Frame) error {
 	var d templateData
 	for f, covered := range fileSet(covered, uncovered) {
-		remain := filepath.Clean(strings.TrimPrefix(f, prefix))
+		if strings.HasPrefix(f, rg.buildDir) {
```

Comment:
Should this be `!HasPrefix`?
We try to remove rg.buildDir prefix, but if we have it, we return an error here...
This trings.HasPrefix should always be true. Or am I missing something?
Please try to generate a report for netbsd locally.

Reply:
Or it may be easier to use tools/syz-cover/syz-cover.go
You can create a small raw cover file with few real PCs and then test different flags combinations and also try to move sources and test case when KernelSrc and KernelBuildSrc are different.